src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_RefReadOnly.cs

```diff
@@ -0,0 +1,642 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.CSharp.UnitTests;
+using Roslyn.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using System.Linq;
+using Xunit;
+using Microsoft.CodeAnalysis.CSharp.Symbols.Metadata.PE;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    public class AttributeTests_RefReadOnly : CSharpTestBase
+    {
+        [Fact]
+        public void RefReadOnlyIsWrittenToMetadata_SameAssembly_Method()
+        {
+            var text = @"
+namespace System.Runtime.CompilerServices
+{
+    public class ReadOnlyAttribute : System.Attribute { }
+}
+class Test
+{
+    public ref readonly int M(ref readonly int x) { return ref x; }
+}
+";
+
+            CompileAndVerify(text, verify: false, symbolValidator: module =>
+            {
+                var method = module.ContainingAssembly.GetTypeByMetadataName("Test").GetMethod("M");
+                Assert.Equal(RefKind.RefReadOnly, method.RefKind);
+                Assert.True(method.ReturnsByRefReadonly);
+
+                var parameter = method.GetParameters().Single();
+                Assert.Equal(RefKind.RefReadOnly, parameter.RefKind);
+
+                var parameterAttribute = parameter.GetAttributes().Single().AttributeClass;
+                Assert.Equal("ReadOnlyAttribute", parameterAttribute.MetadataName);
+                Assert.Equal(module.ContainingAssembly.Name, parameterAttribute.ContainingAssembly.Name);
```

Comment:
Will do.

---
In reply to: [110280319](https://github.com/dotnet/roslyn/pull/18358#discussion_r110280319) [](ancestors = 110280319)